### DISCUSSION TITLE
Implementation of "Avoid Toll Road" and nodes barrier=toll_booth ?

### DISCUSSION THREAD

**Discussion by muralito** [organicmaps/organicmaps#3738] [Q&A]
Hi. 

The implementation of "Avoid Toll Road" settings seems to not be taking in account the nodes "barrier=toll_booth".
Is not enough with these nodes? Is that the intended behaviour and should the way also be tagged as "toll=yes"?

**Comment by biodranik** [comment]
Why can't you mark roads with toll=yes according to OSM? https://wiki.openstreetmap.org/wiki/Key:toll

**Reply by muralito** [reply]
I can, but i didn't know it was mandatory. Also in the cases i know, i would just tag a very short length of about 10 meters (under the roof of the toll booth) , because you can go just almost until the toll booth, do an U-turn and return to where you came from without paying anything.

**Reply by biodranik** [reply]
According to OSM wiki, you should tag paid roads only where there is no way to turn back for free. And OM will show a "paid road" sign in that case properly.

**Reply by muralito** [reply]
OK. I will tag that small fragments then...

**Reply by biodranik** [reply]
Why don't you want to tag the whole road?
